app/controllers/posts_controller.rb

```diff
@@ -8,16 +10,28 @@ def show
   end
 
   def create
-    @post = Post.new(post_params) # TODO: author, location
-    @post.likes = 0
-    @post.author = 'garbo'
+    @post = current_user.post.new(post_params) # TODO: author, location
+    @post.author = current_user.email
     @post.save!
 
     redirect_to @post
   end
 
+  def destroy
+    @post.destroy
+    redirect_to '/posts/'
+  end
+
   private
 
+  def find_post
+    @post = Post.find(params[:id])
+  end
+
+  def find_user
+    @user = User.find_by_email(params[current_user.email])
```

Comment:
haven't looked into devise much but can't we just say `@user = current_user` here?